### PR TITLE
Update API, Improve Performance Based On DeepSource Analytics, Format Code, Write Docstrings 

### PR DESCRIPTION
**Pull request by XtremeDevX**

### CODE REVIEW

File: api/src/auth/util/fetchOauthClientInfo.ts [hunk 1]
```diff
@@ -1,6 +1,15 @@
 import { AuthProviderStandaloneType } from "~/types/AuthProvider.type";
 
-export default function fetchOauthClientInfo(provider: AuthProviderStandaloneType) {
+/**
+ * Fetches the OAuth client info from the Auth0 account.
```
[the remark this answers is not in the document]
Comment (fullstackslayer) [reply api/src/auth/util/fetchOauthClientInfo.ts:4]: this function is to return the client id, secret, and callback URL based on the provider name u specify. 

e.g if I pass in "github" ill get back GITHUB_CLIENT_ID, GITHUB_CLIENT_SECRET from the env. just a little hack to simplify stuff. pls refactor the docstring to reflect this description.

File: web/hoc/api/useConn.ts [hunk 2]
```diff
@@ -2,5 +2,5 @@ import { useContext } from "react";
 import { UseApiConnectionContext } from "./UseApiConnection";
 
 export const useConn = () => {
-    return useContext(UseApiConnectionContext).conn!;
-};
\ No newline at end of file
+  return useContext(UseApiConnectionContext).conn as any;
```
Comment (fullstackslayer) [inline web/hoc/api/useConn.ts:5]: why can't we explicitly deem it defined? What's the need of `as any`. the non-null assertion is better practice.

File: api/src/auth/util/discordAvatarUrl.ts [hunk 3]
```diff
@@ -1,3 +1,6 @@
+/**
+ * Gets the Discord avatar URL for the given user ID.
```
Comment (fullstackslayer) [inline api/src/auth/util/discordAvatarUrl.ts:2]: change this to "given user avatar hash"

**Review by fullstackslayer — CHANGES_REQUESTED** [review]

**Comment by XtremeDevX** [comment]
@fullstackslayer can we merge this now

**Review by fullstackslayer — APPROVED** [review]
awesome tysm

**Comment by fullstackslayer** [comment]
time to resolve conflicts 😩

**Pull request merged**